streams/src/test/java/org/apache/kafka/streams/processor/internals/StreamThreadTest.java

```diff
@@ -695,6 +698,85 @@ public void shouldShutdownTaskManagerOnCloseWithoutStart() {
         EasyMock.verify(taskManager);
     }
 
+    @Test
+    public void shouldNotAccessTaskManagerWhenPendingShutdownInRunOnce() {
```

Comment:
I would give this test a more meaningful name, e.g. `shouldNotThrowWhenPendingShutdownInRunOnce`.